### PR TITLE
added controller for truck backer problem

### PR DESCRIPTION
**Pull request by imran3180**
Changes Summary:

1. Added **set_state** method in the truck class. This method will update the internal variables of the truck class and used for drawing the truck state which is output by the emulator.

2. NN for Controller which learns how to park the truck.

### CODE REVIEW

File: 14-truck_backer-upper.ipynb [hunk 1]
```diff
@@ -249,7 +256,7 @@
    "source": [
     "# Build expert data set\n",
     "\n",
-    "episodes = 10\n",
+    "episodes = 100\n",
```
Comment (Atcold) [inline 14-truck_backer-upper.ipynb]: Why changing this? Its only purpose it to show how data is generated.
To generate the training data set you want to use 10,000 episodes.

**Comment by Atcold** [comment]
This is not parking the truck. Also, I'm not sure how you verify the success of the task.
When generating the data and training the controller we do not display the truck:

```python
truck = Truck(); episodes = 10_000
```

which is equivalent to

```python
truck = Truck(display=False); episodes = 10_000
```

So, how is the last cell showing anything? I'm confused.

**Comment by Atcold** [comment]
Haha, this is cute! <3
@imran3180, you gave pushing permission to @zcgeng to your repo, and @zcgeng pushed a new commit on the branch is currently supporting this PR.

**Comment by imran3180** [comment]
No. It's not working properly yet. @zcgeng pushed onto my branch and I didn't know that I have raised PR on the same branch.

My solution still getting stuck at some random places rather than at dock. 

I am closing this PR for now since its creating confusion. I will raise new PR once when I will get a better solution.

**Comment by Atcold** [comment]
Hahaha, okay, people don't know how GitHub works, hahaha.

**Pull request closed without merging**